### PR TITLE
Post about using the new dev-watch remote option for development.

### PR DESCRIPTION
**Pull request by ehsavoie**
Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

### CODE REVIEW

File: _posts/2022-08-15-Remote-dev-watch.adoc [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+title: Remote dev-watch development with WildFly Jar Maven Plugin
+date: 2022-08-15
+tags: wildfly bootable jar bootable-jar openshift cloud remote
+synopsis: Bootable JAR Maven plugin 8.0.0.Alpha1 brings in a new way to make remote development easier.
+author: ehsavoie
+---
+
+The 8.0.0.Alpha1 version of the link:https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[WildFly bootable JAR Maven plugin] has been released.
+
+This  https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/tag/8.0.0.Alpha1release] is not yet Final, as it is only there to gather feedback on a new feature to simplify develoment on the 'cloud' using the `dev-watch` goal.
```
Comment (jfdenise) [inline _posts/2022-08-15-Remote-dev-watch.adoc:12]: Wrongly formatted link
"to simplify develoment" ==> that simplifies development

File: _posts/2022-08-15-Remote-dev-watch.adoc [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+
+For people who are not familiar with WildFly bootable JAR and its `dev-watch` goal, I strongly recommend that you read 
+this link:https://www.wildfly.org/news/2020/12/15/bootable-jar-3.0.Beta-Released/[blog post] that covers it in details. 
+
+In addition to bug fixes, this release brings an evolution of the `dev-watch` goal that makes repackaging of your application no more required.
```
Comment (jfdenise) [inline _posts/2022-08-15-Remote-dev-watch.adoc:18]: This line seems copy paste and not needed,

File: _posts/2022-08-15-Remote-dev-watch.adoc [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+For people who are not familiar with WildFly bootable JAR and its `dev-watch` goal, I strongly recommend that you read 
+this link:https://www.wildfly.org/news/2020/12/15/bootable-jar-3.0.Beta-Released/[blog post] that covers it in details. 
+
+In addition to bug fixes, this release brings an evolution of the `dev-watch` goal that makes repackaging of your application no more required. 
+
+## Dev-watch goal
+
+The current `dev-watch` goal, although offering an efficient workflow to develop WildFly applications, requires the application or server to run locally, in the same place as the project. The improvement made on this release is to allow the application or server to run remotely so that it can be in an environment that is closer to the target runtime environment.
+We are going to use https://github.com/wildfly-extras/wildfly-jar-maven-plugin/tree/main/examples/remote-microprofile-config [this example] to see how we can work remotely.
```
Comment (jfdenise) [inline _posts/2022-08-15-Remote-dev-watch.adoc:23]: A space to remove after microprofile-config

File: _posts/2022-08-15-Remote-dev-watch.adoc [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+[source,bash]
+----
+$ oc create route edge management-remote-microprofile-config-bootable --service=remote-microprofile-config-bootable --port=9990 --insecure-policy='Redirect'
+----
+
+### Develop and update this application
+
+video::Yc3w4OKRkGA[youtube,width=560,height=315,title="YouTube video player"]
+
+Now we need to run the `dev-watch` goal and remotely attach to the Wildfly Managment API.
```
Comment (jfdenise) [inline _posts/2022-08-15-Remote-dev-watch.adoc:112]: Managment ==> Management

File: _posts/2022-08-15-Remote-dev-watch.adoc [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+For this ewe need to execute the follwing command line:
```
Comment (jfdenise) [inline _posts/2022-08-15-Remote-dev-watch.adoc:113]: ewe ==> we
follwing ==> following

**Review by jfdenise — CHANGES_REQUESTED** [review]

File: _posts/2022-09-14-Remote-dev-watch.adoc [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+title: Remote dev-watch development with WildFly Jar Maven Plugin
+date: 2022-09-14
+tags: wildfly bootable jar bootable-jar openshift cloud remote
+synopsis: Bootable JAR Maven plugin 8.0.0.Alpha2 brings in a new way to make remote development easier.
+author: ehsavoie
+---
+
+The 8.0.0.Alpha2 version of the link:https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[WildFly bootable JAR Maven plugin] has been released.
+
+This https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/tag/8.0.0.Alpha2release] is not yet Final, as it is only there to gather feedback on a new feature that simplifies development on the 'cloud' using the `dev-watch` goal.
```
Comment (jmesnil) [inline _posts/2022-09-14-Remote-dev-watch.adoc:12]: https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/tag/8.0.0.Alpha2[release]

Comment (jmesnil) [inline _posts/2022-09-14-Remote-dev-watch.adoc:12]: do you mean "cloud" or `cloud` (the use of single quote looks weird)?

File: _posts/2022-09-14-Remote-dev-watch.adoc [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+For people who are not familiar with WildFly bootable JAR and its `dev-watch` goal, I strongly recommend that you read 
+this link:https://www.wildfly.org/news/2020/12/15/bootable-jar-3.0.Beta-Released/[blog post] that covers it in details. 
+
+## Dev-watch goal
+
+The current `dev-watch` goal, although offering an efficient workflow to develop WildFly applications, requires the application or server to run locally, in the same place as the project. The improvement made on this release is to allow the application or server to run remotely so that it can be in an environment that is closer to the target runtime environment.
```
Comment (jmesnil) [inline _posts/2022-09-14-Remote-dev-watch.adoc:19]: "application or server". What do you mean? What's the application in that context? the deployment?

File: _posts/2022-09-14-Remote-dev-watch.adoc [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+We are going to use https://github.com/wildfly-extras/wildfly-jar-maven-plugin/tree/main/examples/remote-microprofile-config[this example] to see how we can work remotely.
+
+IMPORTANT: This application applies the script `anonymous-management.cli` which disable security on the Management API of WildFly, please make sure not to include it when going to production.
+
+## Developping with a docker container.
+
+video::ESI6EmQy4c8[youtube,width=560,height=315]
```
Comment (jmesnil) [inline _posts/2022-09-14-Remote-dev-watch.adoc:26]: nothing is displayed when I see the file. Will it be generated at publication time?

File: _posts/2022-09-14-Remote-dev-watch.adoc [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+----
+$ curl http://${container.ip.address}:8080
+config1 = Value from Config1 comes from an env var in the DeploymentConfig
+config2 = Value for config2 comes from a properties file inside the application
+config3 = Default value for config3 comes from my code
+----
+
+Once this is done you can edit the code and your changes will be automatically pushed to the remote container.
+For example:
+ * Change the config2 property value to be "Hello from dev-watch remote" in the file: src/main/resources/META-INF/microprofile-config.properties.
```
Comment (jmesnil) [inline _posts/2022-09-14-Remote-dev-watch.adoc:78]: Change the `config2` property value to be "Hello from dev-watch remote" in the `src/main/resources/META-INF/microprofile-config.properties` file.

File: _posts/2022-09-14-Remote-dev-watch.adoc [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+[source,bash]
+----
+$ curl https://$(oc get route remote-microprofile-config-bootable --template='{{ .spec.host }}')
+config1 = Value from Config1 comes from an env var in the DeploymentConfig
+config2 = Hello from dev-watch remote
+config3 = Default value for config3 comes from my code
+----
+
+## To conclude
```
Comment (jmesnil) [inline _posts/2022-09-14-Remote-dev-watch.adoc:154]: ## Conclusion

**Review by jmesnil — CHANGES_REQUESTED** [review]
LGTM except a few cosmetic changes

File: _posts/2022-09-14-Remote-dev-watch.adoc (hunk 7 above)
Comment (ehsavoie) [reply _posts/2022-09-14-Remote-dev-watch.adoc:19]: bootable application or server

File: _posts/2022-09-14-Remote-dev-watch.adoc (hunk 8 above)
Comment (ehsavoie) [reply _posts/2022-09-14-Remote-dev-watch.adoc:26]: I hope so

**Review by jmesnil — APPROVED** [review]

**Pull request merged**